File: app/models/task.py
from pydantic import BaseModel


class TaskRequest(BaseModel):
    title: str
    description: str

# Pydantic model for the response
class TaskResponse(BaseModel):
    title: str
    description: str
    acceptance_criteria: list[str]
    risks: list[str]
    risk_management: list[str]
    dependencies: list[str]
    third_party_apis: list[str]

File: web_server/app/types/task.ts
// types/task.ts
export interface TaskRequest {
    title: string;
    description: string;
  }
  
  export interface TaskResponse {
    title: string;
    description: string;
    acceptance_criteria: string[];
    risks: string[];
    risk_management: string[];
    dependencies: string[];
    third_party_apis: string[];
    error?: string;
  }  